pkg/backend/backend.go

```diff
@@ -316,6 +316,9 @@ type QueryOperation struct {
 
 // StackConfiguration holds the configuration for a stack and it's associated decrypter.
 type StackConfiguration struct {
+	// List of ESC environments imported by the stack being updated.
+	EscEnvironments []string
```

Comment:
Isn't `Environment` below also an ESC Environment? What's the distinction here? That's the resolved value and this is the list of imports?

Reply:
Yah I can see that it's confusing. the `Environments` here is for the config values under `pulumiConfig` from the resolved esc values. I think it might be better if I rename this back to `EnvironmentImports` then